pennylane/ops/op_math/evolve.py

```diff
@@ -0,0 +1,184 @@
+"""
+
+from typing import Union
+
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation, Operator
+
+from .exp import Evolution
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+
+def evolve(op: Union[Operator, ParametrizedHamiltonian]):
+    """Returns a new operator that computes the evolution of the given operator.
+
+    Args:
+        op (Operator): operator to evolve
+
+    Returns:
+        Evolution | ParametrizedEvolution: evolution operator
+    """
+    if isinstance(op, ParametrizedHamiltonian):
+
+        def parametrized_evolution(params: list, t, dt=None):
+            """Constructor for the :class:`ParametrizedEvolution` operator.
+
+            Args:
+                params (list): params (ndarray): trainable parameters
+                t (Union[float, List[float]]): If a float, it corresponds to the duration of the evolution.
+                    If a list of two floats, it corresponds to the initial time and the final time of the
+                    evolution.
+                dt (float, optional): the time step used by the differential equation solver to evolve the
+                    time-dependent Hamiltonian. Defaults to XXX.
+
+            Returns:
+                ParametrizedEvolution: class used to compute the parametrized evolution of the given
+                    hamiltonian
+            """
+            return Evolve(H=op, params=params, t=t, dt=dt)
+
+        return parametrized_evolution
+
+    return Evolution(generator=op, param=1.0)
+
+
+class Evolve(Operation):
+    r"""Parametrized evolution gate.
+
+    For a time-dependent Hamiltonian of the form
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j` it implements the corresponding
+    time-evolution operator :math:`U(t_1, t_2)`, which is the solution to the time-dependent
+    Schrodinger equation
+
+    .. math:: \frac{d}{dt}U(t) = -i H(v, t) U(t).
+
+
+    Under the hood, it is using a numerical ordinary differential equation solver.
+
+    Args:
+        H (ParametrizedHamiltonian): hamiltonian to evolve
+        params (ndarray): trainable parameters
+        t (Union[float, List[float]]): If a float, it corresponds to the duration of the evolution.
+            If a list of two floats, it corresponds to the initial time and the final time of the
+            evolution. Note that such absolute times only have meaning within an instance of ``ParametrizedEvolution`` and will not affect other gates.
+        dt (float): the time step used by the differential equation solver to evolve the
+            time-dependent Hamiltonian. Defaults to XXX.
+        time (str, optional): The name of the time-based parameter in the parametrized Hamiltonian.
+            Defaults to "t".
+        do_queue (bool): determines if the scalar product operator will be queued. Default is True.
+        id (str or None): id for the scalar product operator. Default is None.
+    """
+
+    _name = "ParametrizedEvolution"
```

Comment:
I'm confused :laughing: Is it `Evolve` or `ParametrizedEvolution`?

Reply:
I don't think so. Moreover, the PL documentation says to avoid punctuation when the sentence is small:
<img width="833" alt="image" src="https://user-images.githubusercontent.com/49232747/213130175-af8aab7e-48c9-41fb-8d37-c5aabf204920.png">

Reply:
`Evolve`! Thanks for the catch 😄